### ISSUE TITLE
Make tree fully accessible (implement tree view application role pattern) [34]

### ISSUE THREAD

**Issue by interactivellama** [salesforce/design-system-react#609]
From @SiTaggart, this is the primary source of truth of what show a tree should function:
http://accessibleculture.org/articles/2013/02/not-so-simple-aria-tree-views-and-screen-readers/
http://w3c.github.io/aria/practices/aria-practices.html#TreeView
- Links within nodes and expand buttons have an implicit `tabindex=0` currently. When the tree is made to use the tree application pattern (ARIA role), then all tabbing should be removed ( `tabindex=-1`).
- Current state of the tree is the ARIA tree pattern is not present due to delays in figuring the API for the first nested data controlled component in the library.
- Tree may be a good candidate for the `keyboardNavigable` mixin that is used in MenuDropdown.
- `aria-activedescendant` should be the node id that has focus. This is used when a composite widget is responsible for managing its current active child to reduce the overhead of having all children be focusable. Examples include: multi-level lists, trees, and grids. In some implementations the user agent may use aria-activedescendant to tell assistive technologies that the active descendant has focus. Authors MAY use the aria-activedescendant attribute on the focused descendant of a composite widget; for example, on a textbox descendant of a combo box.
- Having a focus state, should require making the tree component having state and therefore use `React.createClass`.

Additional a11y links for reference (however I'm not sure all the information is accurate):
- http://www.oaa-accessibility.org/examplep/treeview1/
- http://w3c.github.io/aria/practices/aria-practices.html#TreeView
- https://www.w3.org/WAI/GL/wiki/Using_aria-activedescendant_to_allow_changes_in_focus_within_widgets_to_be_communicated_to_Assistive_Technology
  https://www.marcozehe.de/2012/02/06/if-you-use-the-wai-aria-role-application-please-do-so-wisely/

Please review https://github.com/salesforce-ux/design-system-react/blob/master/CONTRIBUTING.md#concepts-and-best-practices for a general overview of concepts used in this library.

**Comment by interactivellama** [comment]
@chriscorwin Are there any updates on this issue?
